packages/graph-editor/src/components/controls/array.tsx

```diff
@@ -0,0 +1,178 @@
+import { observer } from 'mobx-react-lite';
+import { JSONTree } from 'react-json-tree';
+import React from 'react';
+import { IField } from './interface';
+import { toJS } from 'mobx';
+import { Button, IconButton, Select, Stack, TextInput } from '@tokens-studio/ui';
+import { Minus, Plus } from 'iconoir-react';
+import { ColorPickerPopover } from '../colorPicker';
+import { ANY, AllSchemas, COLOR, Input, NUMBER, STRING } from '@tokens-studio/graph-engine';
+import { delayedUpdateSelector } from '@/redux/selectors';
+import { useSelector } from 'react-redux';
+
+const inputItemTypes = [STRING, NUMBER, COLOR];
+
+export const ArrayField = observer(({ port }: IField) => {
+    const [value, setValue] = React.useState(port.value);
+    const [selectItemsType, setSelectItemsType] = React.useState(port.type.items.$id);
+
+    const [autofocusIndex, setAutofocusIndex] = React.useState<number | null>(null);
+    const useDelayed = useSelector(delayedUpdateSelector);
+
+    React.useEffect(() => {
+        setValue(port.value);
+    }, [port.value]);
+
+    React.useEffect(() => {
+        if (autofocusIndex !== null) {
+            setAutofocusIndex(null);
+        }
+    }, [autofocusIndex]);
+
+    const portName = port.constructor.name;
+    const itemsType = port.type.items.$id;
```

Comment:
We should draw a distinction between the underlying static type and the runtime type . You probably want the `fullType` in order to do this

Reply:
`fullType()` is different in this case. I think we need the dynamic type here